File: typescript-recruitment-test/src/plc.ts
import { Category, DataType } from './types';

export const loadPlcTree = (filePath: string): Category => {
    // TODO: implements and replace the following line
    return {name: 'root', categories: [], measures: []};
};

/**
 *
 * @param category : category to search in
 * @param searchingString : string to search in measure name
 */
export const searchByName = (category: Category, searchingString: string): void => {
    // TODO
};

/**
 *
 * @param category : category to search in
 * @param dataType : dataType to search in measure
 */
export const searchByDataType = (category: Category, dataType: DataType): void => {
    // TODO
};

File: java-recruitment-test/src/main/java/com/scorpio/Plc.java
package com.scorpio;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class Plc {

    /**
     * main function of the application
     *
     * @param args
     */
    public static void main(String[] args) {
        Plc plc = new Plc();
        Category category = plc.loadPlcTree(System.getProperty("user.dir") + "/resources/plcTree.json");
        category.searchByName("Piece-counter");
        category.searchByDataType(DataType.BOOLEAN);
    }

    /**
     * Load the data tree of the Plc from a json file
     *
     * @param filePath - the tree file path
     * @return the root category of the tree
     */
    Category loadPlcTree(String filePath) {
       // TODO
        return null;
    }
}
